recipe-generator-node/src/index.js

```diff
@@ -0,0 +1,118 @@
+const { ArgumentParser } = require("argparse");
+const { exec } = require("child_process");
+const http = require("http");
+const fs = require("fs");
+
+const RECIPE_VERSION = "0.3.4";
+const GENERATOR_URL = `http://search.maven.org/remotecontent?filepath=ca/derekcormier/recipe/recipe-generator/0.3.4/recipe-generator-${RECIPE_VERSION}-jar-with-dependencies.jar`;
+
+main();
+
+async function main() {
+    const options = parseArgs();
+    try {
+        await downloadGeneratorJar(options);
+    }
+    catch(err) {
+        console.log(err);
+    }
+    runGeneration(options);
+}
+
+function parseArgs() {
+    const parser = new ArgumentParser({
+        addHelp: true
+    });
+    parser.addArgument("domain", {
+        help: "name of ingredient domain",
+        type: "string"
+    });
+    parser.addArgument("flavour", {
+        help: "type of generation to perform; valid options: java-ingredient, java-hook, js-ingredient, ts-ingredient",
+        type: "string"
+    });
+    parser.addArgument("cookbook", {
+        help: "path to the yaml cookbook definition file",
+        type: "string"
+    });
+    parser.addArgument("targetDir", {
+        help: "directory to output generated files",
+        type: "string"
+    });
+    parser.addArgument("--javaPackage", {
+        help: "java package for generated classes",
+        type: "string",
+        metavar: "package"
+    });
+    parser.addArgument("--ingredientPostfix", {
+        help: "string to append to ingredient names",
+        type: "string",
+        metavar: "postfix"
+    });
+    parser.addArgument("--jarDownloadDir", {
```

Comment:
Why not put it in a subdir of where this npm package is installed? Or could even ship it as part of the npm package... What's better/worse: an npm package that is a lot of megs or an npm package that downloads a lot of megs on first-run? 🤔

Reply:
Good question. If I published the jar in every npm package I'd be taking up a lot of their space, especially given that I use the same versions for all of the sub-projects: sometimes I need to publish no-op packages if only one of them are updated just to keep it consistent.

Reply:
As for where I put it, I'm pretty sure that at runtime the cwd for this script is where it was executed, not the package location in `node_modules`, but I'll have to double-check that.

Reply:
Or I think you can also have a `postinstall` in the package.json `scripts` to run stuff like this... it could download the jar from mvnwhatever.org as part of the "install" of this package. That might be the "correct" way of doing it now that I think about it more...